### PR TITLE
Add sending custom messages to PX4 from MAVROS

### LINKED ISSUE #806: Sending custom message from MAVROS to PX4 documentation example
I apologise if this is not the correct place but I found a great example that could be added to the official documentation. It can be seen in [this link](https://github.com/JoonmoAhn/Sending-Custom-Message-from-MAVROS-to-PX4/issues/1#issue-252489957). I went through it and everything is running smoothly.

It isn't exactly copy paste as some code needs to change as it is almost 2 years old but ~90% of it is plug and play and the rest is only a few minor changes.

### PR DESCRIPTION
**Pull request by hamishwillee**
Kindly contributed by @JoonmoAhn in https://github.com/JoonmoAhn/Sending-Custom-Message-from-MAVROS-to-PX4/issues/1

This has not yet been updated to recent versions.

Fixes #806 

@RicardoM17 can you update to the current dev environment? You can edit this branch directly?

### CODE REVIEW

**Comment by RicardoM17** [comment]
Hey @hamishwillee ! Sorry I've been away the last couple of days 😎 🌞 

Yap will do! I'll get to it in my off times during this week. 👍

**Comment by hamishwillee** [comment]
@RicardoM17 Great! Are you happy to edit this branch directly or do you need me to submit and you do a separate PR?

**Comment by RicardoM17** [comment]
Hey @hamishwillee I'm fine with whatever is easiest for you. No real preference from my side.

File: en/ros/mavros_custom_messages.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+   ```
+
+Now you are ready to build all your work!
+
+## Building
+
+### Build for ROS
+
+1. In your workspace enter: `catkin build`.
+1. Beforehand, you have to set your "px4.launch" in (/workspace/src/mavros/mavros/launch).
```
Comment (mrivi) [inline en/ros/mavros_custom_messages.md:286]: This part doesn't have anything to do with creating a new mavros plugin.. It's how to connect mavros and the flight controller

File: en/ros/mavros_custom_messages.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+   ```xml
+   ...
+     <arg name="fcu_url" default="/dev/ttyACM0:57600" />
+     <arg name="gcs_url" default="udp://:14550@xxx.xx.xxx.xxx:14557" />
+   ...
+   ```
+
+### Build for PX4
+
+1. Build your firmware
```
Comment (mrivi) [inline en/ros/mavros_custom_messages.md]: again.. this is how you build and flash the Firmware generally

File: en/ros/mavros_custom_messages.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+       { };
+
+       void initialize(UAS &uas_)
+       {
+           uas = &uas_;
+           keyboard_sub = nh.subscribe("keyboard_sub", 10, &KeyboardCommandPlugin::keyboard_cb, this);
+       };
+
+       const message_map get_rx_handlers() {
+           return {/* RX disabled */ };
```
Comment (mrivi) [inline en/ros/mavros_custom_messages.md:41]: To be a complete example on how to create a mavros plugin, there could be a general exaple on how to sends a message from the FCU to ROS and how to create a custom mavros message

File: en/ros/mavros_custom_messages.md [hunk 4]
```diff
@@ -0,0 +1,348 @@
+# Sending Custom Message from MAVROS to PX4
+
+> **Warning** This article is not compatible with current development environment/stable versions. 
+  It has been tested against:
+  - **Ubuntu:** 14.04
```
Comment (mrivi) [inline en/ros/mavros_custom_messages.md:5]: I have done it before for 16.04 and Melodic + 18.04. The steps are the same so this guide should generalize well

Comment (hamishwillee) [reply en/ros/mavros_custom_messages.md:5]: Thanks @mrivi , @RicardoM17 is going to update so can remove this caveat (or more likely replace with latest versions when he does so)

Ricardo, please just add your changes directly into this branch/PR.

File: en/ros/mavros_custom_messages.md (hunk 2 above)
Comment (hamishwillee) [reply en/ros/mavros_custom_messages.md]: Thanks. I've made this general

File: en/ros/mavros_custom_messages.md (hunk 1 above)
Comment (hamishwillee) [reply en/ros/mavros_custom_messages.md:286]: @RicardoM17 Note these suggestions. If you have questions, I can help address.

**Comment by RicardoM17** [comment]
@hamishwillee I've started working on this. Didn't have much time so only did the mavros part however, it seems that I can't push to this branch directly.

I've forked your branch and I will be adding my modifications there in the meantime. [Link](https://github.com/RicardoM17/Devguide/tree/ros_custom_message).

Let me know how you want to proceed ;)

Once I find time I'll keep on going. Been quite busy :sweat_smile:

**Comment by hamishwillee** [comment]
Hi @RicardoM17 

I've merged this PR. The note up the top makes it clear that it hasn't been validated yet

> Let me know how you want to proceed ;)

So now can you push a follow on PR, just linking to this one. If you think you might be short on time but are doing meaningful chunks each time let me know, and I will merge the PRs "in bits".

Cheers
Hamish

**Comment by RicardoM17** [comment]
Sounds good @hamishwillee 

Next week I'll see if I have time, will probably end up doing quite a fair chunk. If by the end of the week it's still not complete then we can merge the "half done" PR.

Cheers!

**Pull request merged**